server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1263,34 +1271,26 @@ private static void validateAliasOrIndex(String expression) {
             if (expression.charAt(0) == '_') {
                 throw new InvalidIndexNameException(expression, "must not start with '_'.");
             }
+            return expression;
         }
 
-        private static boolean aliasOrIndexExists(Context context, String expression, boolean throwExceptionIfAbsent) {
+        private static RuntimeException aliasOrIndexExists(Context context, String expression) {
```

Comment:
It's pretty expensive to raise an exception, especially if you are not using it. Would it be possible to wrap the RuntimeException in a supplier and lazily raise the exception  ?

Reply:
I have pushed [bbcce03](https://github.com/elastic/elasticsearch/pull/90902/commits/bbcce03d9d2ad4f2039f5067d512c2c22ecee05b), I think this addresses your concern.